src/main/java/com/redhat/composer/config/retriever/embeddingmodel/NomicLocalEmbeddingModelClient.java

```diff
@@ -13,7 +13,7 @@ public class NomicLocalEmbeddingModelClient extends BaseLocalEmbeddingModelClien
 
   private static final String MODEL_FOLDER = "embeddings/nomic/";
 
-
+  // TODO: Don't do work that can fail during static initialization. If it fails, the JVM gives very misleading ClassDefNotFoundException
```

Comment:
I am not sure there is another way to load this model directly. 

But yeah I would generally like to find a better way in which to do this. Specifically for nomic they added a direct integration here that I think we could use: https://docs.langchain4j.dev/integrations/embedding-models/nomic

Reply:
I'll move this to the class constructor so that the model and tokenizer are loaded during object initialization, not during classloading. Since it is marked as a `@Singleton`, an instance of the class will only be constructed once so the model loading will still only happen once.

We can change to the direct integration as part of a later effort.